infrastructure/lib/constructs/lambda-fn.ts

```diff
@@ -0,0 +1,219 @@
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as iam from "@aws-cdk/aws-iam";
+import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as lambdaNodeJs from "@aws-cdk/aws-lambda-nodejs";
+import * as cdk from "@aws-cdk/core";
+import * as s3 from "@aws-cdk/aws-s3";
+import { HttpMethod } from "../http";
+import * as dynamodb from "@aws-cdk/aws-dynamodb";
+import { SecureBucket } from "./compliant-resources";
+import * as sqs from "@aws-cdk/aws-sqs";
+import { SqsEventSource } from "@aws-cdk/aws-lambda-event-sources";
+import * as sfn from "@aws-cdk/aws-stepfunctions";
+
+/**
+ * An IAM service principal for the API Gateway service, used to grant Lambda
+ * invocation permissions.
+ */
+const APIGW_SERVICE_PRINCIPAL = new iam.ServicePrincipal("apigateway.amazonaws.com");
+
+export interface ApiFunctionPropstest {
+  /**
+   * The HTTP method this route applies to.
+   */
+  readonly method?: HttpMethod;
+
+  /**
+   * The local path where the handler file is located.
+   */
+  readonly handlerPath?: string;
+
+  /**
+   * Any additional props to apply to the NodeJsFunction.
+   *
+   * Additional prop values specified here will override the defaults, including
+   * the `entry` and `environment` if one is provided in this struct. Be careful.
+   */
+  readonly functionPropsOverride?: lambdaNodeJs.NodejsFunctionProps;
+
+  /**
+   * The VPC where resources should be created
+   */
+  readonly lambdaVpc?: ec2.IVpc;
+
+  /**
+   * A prop to determine if the underlying Lambda function has read permissions
+   * for secrets managers.
+   */
+  readonly smtpSecrets?: secretsmanager.ISecret;
+
+  /**
+   * The DynamoDB table where data is stored.
+   */
+  readonly table?: dynamodb.ITable;
+
+  /**
+   * Optional table permissions to grant to the Lambda function.
+   * One of the following may be specified: "ALL", "READ", "READ_WRITE", "WRITE".
+   *
+   * @default - Read/write access is given to the Lambda function if no value is specified.
+   */
+  readonly tablePermissions?: string;
+
+  /**
+   * Secure bucket for S3 functions
+   */
+  readonly bucket?: SecureBucket;
+
+  /**
+   * Optional bucket permissions to grant to the Lambda function.
+   * One of the following may be specified: "READ", "PUT", "DELETE".
+   *
+   * @default - Read access is given to the Lambda function if no value is specified.
+   */
+  readonly bucketPermissions?: string;
+
+  /**
+   * The SQS queue where messages are sent
+   */
+  readonly queue?: sqs.IQueue;
+
+  /**
+   * Optional bucket permissions to grant to the Lambda function.
+   * One of the following may be specified: "SEND", "CONSUME".
+   *
+   * @default - Read access is given to the Lambda function if no value is specified.
+   */
+  readonly queuePermissions?: string;
```

Comment:
Again. Strings should not hold enumerated values.

Reply:
Type should be `QueuePermissions`